src/main.rs

```diff
@@ -33,98 +34,110 @@ mod template;
 type Feature<'a> = &'a str;
 type PackageName<'a> = &'a str;
 type RootFeature<'a> = (PackageName<'a>, Feature<'a>);
+type Result<T> = std::result::Result<T, Error>;
 
 const VERSION_ATTRIBUTE_NAME: &str = "cargo2nixVersion";
 
 fn main() {
     let args: Vec<String> = std::env::args().collect();
     let args: Vec<&str> = args.iter().map(AsRef::as_ref).collect();
+    if let Err(err) = try_main(&args) {
+        eprint!(colorify!(red_bold: "error: "));
+        eprintln!("{}", err);
+        std::process::exit(1);
+    }
+}
 
+fn try_main(args: &[&str]) -> Result<()> {
     match &args[1..] {
         ["--stdout"] | ["-s"] => generate_cargo_nix(io::stdout().lock()),
         ["--file"] | ["-f"] => write_to_file("Cargo.nix"),
         ["--file", file] | ["-f", file] => write_to_file(file),
         ["--help"] | ["-h"] => print_help(),
-        ["--version"] | ["-v"] => println!("{}", version()),
+        ["--version"] | ["-v"] => {
+            println!("{}", version());
+            Ok(())
+        }
         [] => print_help(),
         _ => {
             println!("Invalid arguments: {:?}", &args[1..]);
             println!("\nTry again, with help: \n");
-            print_help();
+            print_help()
         }
     }
 }
 
 fn version() -> Version {
-    Version::parse(env!("CARGO_PKG_VERSION")).expect("parse CARGO_PKG_VERSION")
+    // Since `CARGO_PKG_VERSION` is provided by Cargo itself, which uses the same `semver` crate to
+    // parse version strings, the `unwrap()` below should never fail.
+    Version::parse(env!("CARGO_PKG_VERSION")).unwrap()
 }
 
-fn read_version_attribute(path: &Path) -> Version {
-    let file = fs::File::open(path).expect(&format!("Couldn't open file {}", path.display()));
+fn read_version_attribute(path: &Path) -> Result<Version> {
+    let file = fs::File::open(path).context(format!("Couldn't open file {}", path.display()))?;
```

Comment:
We're ignoring the cause of the failure here. So maybe, we should use `with_context(|e| ..)`?

Reply:
We didn't print the cause in the original code either, but now that you mention it, I think we probably should for `File::open()`. I'll add it in a subsequent commit! 👍

Reply:
`context` already records the error cause in the newly returned error. The recommended way to print the whole error chain is to use [`iter_chain`](https://docs.rs/failure/0.1.6/failure/struct.Error.html#method.iter_chain), similar to what Cargo [does](https://docs.rs/cargo/0.42.0/src/cargo/util/errors.rs.html#403-409). In fact I think we should reuse that function from Cargo.

Reply:
@trha That's a great idea! I'll update the latest commit to do just that.